From af0f7cfdce0c190ed17562efd713a14662eab1eb Mon Sep 17 00:00:00 2001
From: Manish Dharanenthiran <quic_mdharane@quicinc.com>
Date: Mon, 12 Aug 2024 10:21:12 +0530
Subject: [PATCH] hostapd: Remove TESTING_OPTIONS compile flag dependency

In-order to leverage the existing ML link removal support for offloaded
drivers remove the CONFIG_TESTING_OPTIONS compile flag dependency.

Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 hostapd/ctrl_iface.c    | 4 ----
 src/ap/hostapd.c        | 5 -----
 src/ap/hostapd.h        | 2 --
 src/ap/ieee802_11_eht.c | 4 ----
 4 files changed, 15 deletions(-)

--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -4411,7 +4411,6 @@ static int hostapd_ctrl_iface_disable_ml
 }
 
 
-#ifdef CONFIG_TESTING_OPTIONS
 static int hostapd_ctrl_iface_link_remove(struct hostapd_data *hapd, char *cmd,
 					  char *buf, size_t buflen)
 {
@@ -4432,7 +4431,6 @@ static int hostapd_ctrl_iface_link_remov
 
 	return ret;
 }
-#endif /* CONFIG_TESTING_OPTIONS */
 #endif /* CONFIG_IEEE80211BE */
 
 
@@ -5342,12 +5340,10 @@ static int hostapd_ctrl_iface_receive_pr
 	} else if (os_strcmp(buf, "DISABLE_MLD") == 0) {
 		if (hostapd_ctrl_iface_disable_mld(hapd->iface))
 			reply_len = -1;
-#ifdef CONFIG_TESTING_OPTIONS
 	} else if (os_strncmp(buf, "LINK_REMOVE ", 12) == 0) {
 		if (hostapd_ctrl_iface_link_remove(hapd, buf + 12,
 						   reply, reply_size))
 			reply_len = -1;
-#endif /* CONFIG_TESTING_OPTIONS */
 #endif /* CONFIG_IEEE80211BE */
 	} else {
 		os_memcpy(reply, "UNKNOWN COMMAND\n", 16);
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -427,7 +427,6 @@ static int hostapd_broadcast_wep_set(str
 
 
 #ifdef CONFIG_IEEE80211BE
-#ifdef CONFIG_TESTING_OPTIONS
 
 static void hostapd_link_remove_timeout_handler(void *eloop_data,
 						void *user_ctx)
@@ -466,9 +465,6 @@ int hostapd_link_remove(struct hostapd_d
 		   hapd->mld_link_id, count);
 
 	hapd->eht_mld_link_removal_count = count;
-	hapd->eht_mld_bss_param_change++;
-	if (hapd->eht_mld_bss_param_change == 255)
-		hapd->eht_mld_bss_param_change = 0;
 
 	eloop_register_timeout(0, TU_TO_USEC(hapd->iconf->beacon_int),
 			       hostapd_link_remove_timeout_handler,
@@ -478,7 +474,6 @@ int hostapd_link_remove(struct hostapd_d
 	return 0;
 }
 
-#endif /* CONFIG_TESTING_OPTIONS */
 #endif /* CONFIG_IEEE80211BE */
 
 
@@ -634,11 +629,9 @@ void hostapd_free_hapd_data(struct hosta
 
 #ifdef CONFIG_IEEE80211AX
 	eloop_cancel_timeout(hostapd_switch_color_timeout_handler, hapd, NULL);
-#ifdef CONFIG_TESTING_OPTIONS
 #ifdef CONFIG_IEEE80211BE
 	eloop_cancel_timeout(hostapd_link_remove_timeout_handler, hapd, NULL);
 #endif /* CONFIG_IEEE80211BE */
-#endif /* CONFIG_TESTING_OPTIONS */
 
 #endif /* CONFIG_IEEE80211AX */
 }
--- a/src/ap/hostapd.h
+++ b/src/ap/hostapd.h
@@ -562,9 +562,7 @@ struct hostapd_data {
 	 * total, additional 7 characters required. */
 	char ctrl_sock_iface[IFNAMSIZ + 7 + 1];
 
-#ifdef CONFIG_TESTING_OPTIONS
 	u8 eht_mld_link_removal_count;
-#endif /* CONFIG_TESTING_OPTIONS */
 #endif /* CONFIG_IEEE80211BE */
 
 #ifdef CONFIG_NAN_USD
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -800,7 +800,6 @@ out:
 
 static u8 * hostapd_eid_eht_reconf_ml(struct hostapd_data *hapd, u8 *eid)
 {
-#ifdef CONFIG_TESTING_OPTIONS
 	struct hostapd_data *other_hapd;
 	u16 control;
 	u8 *pos = eid;
@@ -865,9 +864,6 @@ static u8 * hostapd_eid_eht_reconf_ml(st
 
 	wpa_hexdump(MSG_DEBUG, "MLD: Reconfiguration ML", eid, eid[1] + 2);
 	return pos;
-#else /* CONFIG_TESTING_OPTIONS */
-	return eid;
-#endif /* CONFIG_TESTING_OPTIONS */
 }
 
 
